### DISCUSSION TITLE
pm:cut - how can I get the resulting geojson after a cut operation

### DISCUSSION THREAD

**Discussion by udos** [geoman-io/leaflet-geoman#1535] [Questions & Help] [answered]
hi,

I'm struggling with getting the resulting geojson after a cut operation.

let me explain:
1. I have a featurecollection with 8 features (linestrings)
2. I cut one of the linestring, so the result is 2
3. the leaflet map shows 9 features
4. I have a pm:cut event listener where I can inspect the resulting 2 linestrings from the operation

so far I was unable to access the resulting featurecollection :|

any ideas/suggestion? thanks 🙏

**Comment by Falke-Design** [comment]
Hi,
in the `pm:cut` event you get the resulting layer in the payload. Just convert it to GeoJSON:

```
	e.layer.on('pm:cut', function(x) {
		console.log('cut geojosn', x.layer.toGeoJSON())
	});
```
https://plnkr.co/edit/9kwkOowYzay8RJVS

**Comment by udos** [comment]
thx. I added numbers to my original question to better explain.

yes, for the "impacted" linestring of the original featurecollection (point 1.) which is now 2 I can do that (in point 4.).

but how can I get to the geojson of the changed  featurecollection which now has 9 linestrings?

**Reply by Falke-Design** [reply, marked as answer]
It is important to understand that Leaflet doesn't work with FeatureCollections. Leaflet has Layers which can be in mutliple LayerGroups and on the map. 
When you are adding your FeatureCollection to the map with [L.GeoJSON](https://leafletjs.com/examples/geojson/), Leaflet converts the FeatureCollection to a LayerGroup and each Feature to a Layer. But for Geoman it doesn't matter if it is in a LayerGroup, only the edited Layer will be changed and for actions like cutting a new Layer is created. It is now your job, to add this new created Layer after cutting to a LayerGroup if you want to generate a FeatureCollection out of it.
To short this up: You need to add the result layer from cutting to the initial L.GeoJSON-LayerGroup and then  call on it `.toGeoJSON()`:

```
const layergroup = L.geoJSON(data);

map.on('pm:cut', function(x) {
	layergroup.addLayer(x.layer);
	console.log(layergroup.toGeoJSON())
});
```

Alternatively you can set the option to add all new layers of drawing / cutting to a specific LayerGroup:
```
map.pm.setGlobalOptions({ layerGroup: layergroup });
```

**Reply by udos** [reply]
thank you very much florian 🙏
_note: I am using leaflet-vue and will use a leflet group to add the single features to it and try as you showed in your example._
